File: Scripts/ParseShortAnswer.swift
//
//  ParseShortAnswer.swift
//  ParseShortAnswer
//
//  Created by 夏中洋 on 2017/9/29.
//  Copyright © 2017年 Xia Zhongyang. All rights reserved.
//

import Foundation

let folderPath = "./"
let csvPath = folderPath + "ShortAnswer.csv"
let shortAnswerString = try String(contentsOfFile: csvPath, encoding: String.Encoding.utf8)
let shortAnswerArray = shortAnswerString.components(separatedBy: "\n")

var i = 0
for line in shortAnswerArray {
    i += 1
    let lineArray = line.components(separatedBy: ",")
    let question = lineArray[0]
    let answer = lineArray[1]
    let configString = "\(question)\n\n\(answer)\nN/A\nN/A\nN/A\n-1"
    try configString.write(toFile: "\(folderPath)/data/sq/\(i).config", atomically: false, encoding: String.Encoding.utf8)
}

File: scripts/ParseShortAnswer.py
import csv
import utils


utils.mkdir("../data/sq")
file_name = utils.ask_file_name()
if file_name:
    with open(file_name, encoding="utf8") as file:
        reader = csv.reader(file)
        i = 0
        for line in reader:
            i += 1
            question = line[0]
            answer = line[1]
            configString = "{0}\n\n{1}\nN/A\nN/A\nN/A\n-1".format(question, answer)
            with open("../data/sq/{}.config".format(i), 'w', encoding="utf8") as config_file:
                config_file.write(configString)
